main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -50,9 +52,17 @@ public void validate(NoticeContainer noticeContainer) {
   }
 
   /** Generates a hash based on route names, type and agency. */
-  private static int getRouteKey(GtfsRoute route) {
-    return Objects.hash(
-        route.routeLongName(), route.routeShortName(), route.routeType(), route.agencyId());
+  private static long getRouteKey(GtfsRoute route) {
+    return HASH_FUNCTION
+        .newHasher()
+        .putUnencodedChars(route.routeLongName())
+        .putChar('\0')
+        .putUnencodedChars(route.routeShortName())
+        .putChar('\0')
+        .putInt(route.routeType().getNumber())
+        .putUnencodedChars(route.agencyId())
```

Comment:
Should `.putChar('\0')` be used here?

```suggestion
        .putChar('\0')
        .putUnencodedChars(route.agencyId())
```

Reply:
No, since this is the last string. A delimiter needs to be placed between a string and the next item.

Reply:
I get it :)